src/core/load-fragment.c

```diff
@@ -5217,9 +5217,15 @@ int config_parse_private_tmp(
         assert(filename);
         assert(rvalue);
 
+        PrivateTmp t = private_tmp_from_string(rvalue);
+        if (t >= 0) {
+                c->private_tmp = t;
+                return 0;
+        }
+
         r = parse_boolean(rvalue);
         if (r < 0) {
```

Comment:
please instead replace use of DEFINE_STRING_TABLE_LOOKUP() by DEFINE_STRING_TABLE_LOOKUP_WITH_BOOLEAN()

and then use a generic enum parser routine here